apps/insights/src/hooks/use-price-formatter.ts

```diff
@@ -1,40 +1,38 @@
 import { useCallback, useMemo } from "react";
 import { useNumberFormatter } from "react-aria";
 
-export const usePriceFormatter = () => {
+export const usePriceFormatter = (exponent?: number) => {
+  // Calculate the number of decimal places based on the exponent
+  // The exponent represents the power of 10, so -8 means 8 decimal places
+  const decimals = exponent === undefined ? undefined : Math.abs(exponent);
+
   const bigNumberFormatter = useNumberFormatter({ maximumFractionDigits: 2 });
   const smallNumberFormatter = useNumberFormatter({
     maximumSignificantDigits: 6,
   });
+  const exponentBasedFormatter = useNumberFormatter({
+    minimumFractionDigits: decimals,
+    maximumFractionDigits: decimals,
+  });
+
   const format = useCallback(
-    (n: number) =>
-      n >= 1000
-        ? bigNumberFormatter.format(n)
-        : formatToSubscriptNumber(smallNumberFormatter.format(n)),
-    [bigNumberFormatter, smallNumberFormatter],
+    (n: number) => {
+      // If we have an exponent, use exponent-based formatting
+      if (decimals !== undefined) {
+        return exponentBasedFormatter.format(n);
+      }
+      // Otherwise, fall back to the old behavior
+      if (n >= 1000) {
+        return bigNumberFormatter.format(n);
+      }
+      return smallNumberFormatter.format(n);
+    },
+    [
+      bigNumberFormatter,
+      smallNumberFormatter,
+      exponentBasedFormatter,
+      decimals,
+    ],
   );
   return useMemo(() => ({ format }), [format]);
 };
-
-const formatToSubscriptNumber = (numString: string) => {
```

Comment:
I think you should consider keeping this formatter, for some cryptos the pricing & CI are very very small fractions and you can have very long streams of zeroes which are very hard to read.  This formatter resolves that issue quite nicely and can be applied on top of the `exponentBasedFormatter`.

<img width="1254" height="541" alt="screenshot-2025-10-17-095310" src="https://github.com/user-attachments/assets/d35c56c5-65b5-4add-8df6-79b2032cfeb3" />